android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -2371,8 +2371,9 @@ public void onTrackRecordingCancelled()
   public void onTrimMemory(int level)
   {
     super.onTrimMemory(level);
-    Logger.d(TAG, "trim memory, level = " + level);
-    if (level >= TRIM_MEMORY_RUNNING_LOW)
+
+    Logger.d(TAG, "Trim memory, level = " + level);
+    if (level >= TRIM_MEMORY_RUNNING_LOW && level != TRIM_MEMORY_UI_HIDDEN)
```

Comment:
What are the benefits/why? If UI is not visible => OM is in the background => no need to keep the memory allocated, because the system will likely kill OM in such case.